servicetalk-http-router-predicate/src/main/java/io/servicetalk/http/router/predicate/InOrderRouter.java

```diff
@@ -67,7 +68,10 @@
                 StreamingHttpService service = pair.service();
                 final HttpExecutionStrategy strategy = pair.routeStrategy();
                 if (strategy != null) {
-                    service = strategy.offloadService(ctx.executionContext().executor(), service);
+                    Executor executor = null == strategy.executor() ?
```

Comment:
on the subject of removing `Executor` from `HttpExecutionStrategy` ... looks like the routers are leveraging this to pass the `Executor` + strategy maybe overriden on the route level (e.g. `RouteExecutionStrategy`, `RouteExecutionStrategyUtils`, ..., `JerseyRouteExecutionStrategyUtils#getRouteExecutionStrategy`, `EndpointEnhancingRequestFilter#defineEndpoint`). Does this impact plans to simplify the `HttpExecutionStrategy` interface? I'm not even sure we need to support specifying different executors on the route level as we apply capacity management independent of the Executor.

Reply:
TBH, I am struggling with exactly that issue in #1695 at the moment. Some of what the routers expect can be handled by using the router executor as the context executor.

Reply:
discussed offline ... plan is to continue with `HttpExecutionStrategy#executor()` removal and focus on routers specifying strategy portions first (required to be "safe"), and we can later add support to specify an `executor` at the route level if required.